main.py

```diff
@@ -137,26 +140,79 @@ def _validate_validation_status(response: requests.Response) -> bool:
         sys.exit(1)
 
 
-def download_html_report(token: str, request_id: str, payload: Dict[str, Any]):
-    response = _retry_request(
+def download_report(token: str, request_id: str, payload: Dict[str, Any], response_type: str):
```

Comment:
Create a small helper function `_download_report` which accepts `response_type` as a string and you can create 2 another function `download_report_as_html` and `download_report_as_json` and call that `_download_report` function with a particular `response_type`.

I think it will be easier to read it like this.